### ISSUE TITLE
'd3_lw.x' fails with 'cannot interpolate with periodic matrices'

### ISSUE THREAD

**Issue by irrotational** [anharmonic/d3q#21]
For all of the calculations in the 'Examples' directory (Graphene, Silicon, MgO), everything runs fine up until 'd3_lw.x' is invoked. Then I get:

'cannot interpolate with periodic matrices'

**Comment by hyder-rescale** [comment]
Did we ever find the solution for this. I am using intel compilers and the error seems to be coming from mp_base.f90 while deallocating the memory

**Comment by paulatz** [comment]
Yes, you have to specify the "-f 2" for the program d_qq2rr.x or far=2 in the input to d3_q2r.x (I think just the latter is enough, but I'm traveling now and cannot check).
Anyway, it should be fixed already in the git version of the code.

Hth

**Comment by paulatz** [comment]
You can regenerate the file of force constants using the command
d3_q2r.x -f dyn -o mat2R.new -f 2

(if your phonon output files are called dyn0, dyn1.. etc, otherwise change dyn accordingly), this should produce a mat2R.new file which non-periodic.

**Comment by minnie178** [comment]
It is running now. Thanks a lot for the help!

**Comment by gk16159** [comment]
Hi everyone, I am having a problem installing d3_wl.x with qe, it shows d3q.x in bin, but none of the thermal2 binaries have been installed or executed. Please help

**Comment by paulatz** [comment]
Please open a separate issue reporting your exact compilation setup and the full output of make or cmake
